spec/index.md

```diff
@@ -613,6 +621,27 @@ Calls must stay in `replied` or `rejected` long enough for polling users to catc
 
 When asking the IC about the state or call of a request, the user uses the request id (see [Request ids](#request-id)) to read the request status (see [Request status](#state-tree-request-status)) from the state tree (see [Request: Read state](#http-read-state)).
 
+#### Synchronous canister calling {#http-sync-call}
+
+Unlike asynchronous canister calling, synchronous canister calling does not initially require the user to poll the Internet Computer for the status of the request. Instead, the user waits for a certified response from the Internet Computer for the initial call. However, if a certified response is not received (due to a timeout or pre-execution error), the user must poll the Internet Computer for the status. Still, the state transitions and semantics of the call itself are the same as for asynchronous canister calling.
```

Comment:
```suggestion
Unlike asynchronous canister calling, synchronous canister calling does not initially require the user to poll the Internet Computer for the status of the request. Instead, the user waits for a certified response from the Internet Computer for the initial call. However, if a certified response is not received (due to a timeout), the user must poll the Internet Computer for the status of the request. In that case, the state transitions and semantics of the call itself are the same as for asynchronous canister calling.
```

Reply:
I strongly believe that the replica should be allowed to conclude a synchronous call before it times out. E.g. if it is stuck in `processing` for too long. Or even IMHO if it's in the received` state for more than a minute or so (someone could attempt to DoS a single canister with synchronous calls; we don't want a replica maintaining 300k concurrent synchronous calls).

So maybe we should be more explicit about replicas being able to terminate a synchronous call at any point before the actual call has timed out or completed.

Could be as simple as replacing "if a certified response is not received (due to a timeout)" with "if a certified response is not produced in reasonable time".

Reply:
Agreed. It's more explicit about it now.